beslab_genesis.yaml

```diff
@@ -1,54 +1,356 @@
 ---
-#this is the scledown version of the lab
-#OS specific
-# VM configurations
-BESLAB_VM_NAME: BesLab
-BESLAB_VM_OS: ubuntu
-BESLAB_VM_GUI: true
-BESLAB_GUI: ubuntu-desktop
-BESLAB_VAGRANT_BOX: hashicorp/bionic64
-BESLAB_USER: vagrant
-BESLAB_PWD: vagrant
-BESLAB_VAGRANT_HOSTNAME: BeSLab.dev
-BESLAB_VAGRANT_MACHINE_NAME: BesLab
-
-# Code collaboration platforme
-BESLAB_CODE_COLLABORATION_PLATFORM_SCOPE: public
-BESLAB_PLATFORM_URL: https://github.com
-
-# Internal datastores
-BESLAB_DATASTORE_SCOPE: public
-BESLAB_ASSESSMENT_DATASTORE_URL: "{{ BESLAB_PLATFORM_URL }}/Be-Secure/besecure-assessment-datastore"
-BESLAB_OSSPOI_DATASTORE_URL: "{{ BESLAB_PLATFORM_URL }}/Be-Secure/besecure-osspoi-datastore"
-BESLAB_ASSESSMENT_DATASTORE_DIR: "/home/{{ BESLAB_USER }}/besecure-assessment-datastore"
-BESLAB_OSSPOI_DATASTORE_DIR: "/home/{{ BESLAB_USER }}/besecure-osspoi-datastore"
-
-
-# Internal languages, tools, repos
+# Madatory BLIman repo info
+BLIMAN_HOSTED_URL: https://raw.githubusercontent.com
+BLIMAN_NAMESPACE: Be-Secure
+BLIMAN_REPO_URL: https://raw.githubusercontent.com/Be-Secure/BLIman/main
+BLIMAN_VERSION: 0.1.0
+BLIMAN_LAB_URL: https://raw.githubusercontent.com/Be-Secure/BeSLab/main
+BESLAB_CODE_COLLAB_TOOL: gitlab
+
+###########################################################
+##
+##### Genesis file is the configuration file of BeSLab
+##
+###########################################################
+
+# [NOTE]
+### By default all features of lab are disabled.
+### To enable the specific feaure uncomment it.
+#
+
+###########################################################
+###########################################################
+## BeSLab Global Configurations
+###########################################################
+
+# BESLAB_TYPE is type of lab to be installed. 
+# Values: personal, public, private.
+#    personal: Installs all the tools in personal system
+#              No separate code collab platform is used.
+#    public: Does use public code collab platform e.g
+#            github, gitlab etc.
+#    private: Does install private code collab platfrom
+#             on the organization infra.
+########################################################### 
+BESLAB_TYPE: personal
+
+# BESLAB_MODE defines the type of BESLAB deployment
+# Values: host, bare , lite
+#    host: Deploy the lab on a hosted VM launched by
+#          oah-bes-vm and vagrant on windows based system.
+#    bare: Deploy the lab on local system or remote system
+#          using Ansible playbooks.
+#    lite: Deploy lab on local system using only shell 
+#          scripts without use use of Ansible or any VM.
+###########################################################
+BESLAB_MODE: lite
```

Comment:
We are passing it during init.

Reply:
I am planning to provide it through Genesis file only.